sbnana/CAFAna/Core/MultiVar.cxx

```diff
@@ -12,11 +12,71 @@ namespace ana
   {
   }
 
+  //----------------------------------------------------------------------
+  /// Helper for \ref MultiVar2D
+  template<class T> class MultiVar2DFunc
+  {
+  public:
+    MultiVar2DFunc(const _MultiVar<T>& a, const Binning binsa,
+                   const _MultiVar<T>& b, const Binning binsb)
+      : fA(a), fBinsA(binsa),
+        fB(b), fBinsB(binsb)
+    {
+    }
+
+    std::vector<double> operator()(const T* sr) const
+    {
+      std::vector<double> returnVec;
+
+      const std::vector<double> vaVec = fA(sr);
+      const std::vector<double> vbVec = fB(sr);
+
+      if(vaVec.size() != vbVec.size())
+        throw std::length_error("MultiVars need to be same size");
```

Comment:
I'm always suspicious of the reliability of an unhandled exception error message actual ends up on the screen, rather than something very unhelpful. Can you just print this message (bonus points for giving the two sizes) and `abort()`?